### PR TITLE
Add HTTP raw binary request support. Add documentation in binary extension

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

**Comment by dyastremsky** [comment]
Looks good! Would it make sense to add an error test case? Like if the user requests the wrong number of bytes (too many, too little). What's expected behavior and can we verify it with testing?

File: src/servers/http_server.cc [hunk 1]
```diff
@@ -2023,7 +2023,7 @@ HTTPAPIServer::GetInferenceHeaderLength(
     evhtp_request_t* req, int32_t content_length, size_t* header_length)
 {
   // Find Inference-Header-Content-Length in header. If missing set to 0
```
Comment (deadeyegoodwin) [inline src/servers/http_server.cc:2025]: I think this comment is outdated

File: qa/L0_http/http_test.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import sys
+sys.path.append("../common")
+
+import unittest
+import numpy as np
+import test_util as tu
+import requests
+
+
+class HttpTest(tu.TestResultCollector):
```
Comment (deadeyegoodwin) [inline qa/L0_http/http_test.py:37]: You should add a successful test case with BYTE type model. Also add failing test cases for BYTE type but not shape [1], fail if multiple variable-sized dimensions, fail if model has more than one input.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

**Review by deadeyegoodwin — DISMISSED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Review by CoderHam — CHANGES_REQUESTED** [review]
@GuanLuo can you rebase to resolve branch conflicts?

**Review by CoderHam — APPROVED** [review]

**Pull request merged**